vyper/compiler/phases.py

```diff
@@ -144,13 +154,35 @@ def generate_ast(source_code: str, source_id: int) -> vy_ast.Module:
     return vy_ast.parse_to_ast(source_code, source_id)
 
 
+def generate_folded_ast(vyper_module: vy_ast.Module) -> vy_ast.Module:
+    """
+    Perform constant folding operations on the Vyper AST.
+
+    This is the second phase of compilation.
+
+    Arguments
+    ---------
+    vyper_module : vy_ast.Module
+        Top-level Vyper AST node
+
+    Returns
+    -------
+    vy_ast.Module
+        Folded Vyper AST
+    """
+    vyper_module_folded = copy.deepcopy(vyper_module)
+    vy_ast.folding.fold(vyper_module_folded)
+
+    return vyper_module_folded
+
+
 def generate_global_context(
     vyper_module: vy_ast.Module, interface_codes: Optional[InterfaceImports],
 ) -> GlobalContext:
     """
     Generate a contextualized AST from the Vyper AST.
 
-    This is the second phase of compilation.
+    This is the third phase of compilation.
```

Comment:
I really don't think we should have these numbers spelled out like this.

Reply:
Yeah, agreed.  I'll add a single comment that the complier phases execute in the same order as the functions are given.